### ISSUE TITLE
Multiple Instances on OSX

### ISSUE THREAD

**Issue by shivharikulhari** [runelite/runelite#9590]
**Is your feature request related to a problem? Please describe.**
Now when you're playing on windows, its pretty simple, you double click the application and it opens another client, I wish it was the same for OSX.

**Describe the solution you'd like**
Having an option for multiple instance so I don't have to "Duplicate" Runelite everytime I want another client running.

**Describe alternatives you've considered**
The only alternatives to this are either having to duplicate runelite, which consumes additional memory on my storage, or having to a different client like OSBuddy, etc..

**Comment by Nightfirecat** [comment]
I'm not very familiar with OSX, do other clients (vanilla, osbuddy, etc.) have this functionality? Do other (java-based) programs generally have this functionality?

**Comment by deathbeam** [comment]
You can use `open -na /Applications/RuneLite.app`

**Comment by shivharikulhari** [comment]
> I'm not very familiar with OSX, do other clients (vanilla, osbuddy, etc.) have this functionality? Do other (java-based) programs generally have this functionality?

Yes, there's no vanilla client for mac ( at least wasn't there last I checked ). Sadly no, OSBuddy also doesn't support it, its not just OSBuddy, its pretty much for every application, you can't run multiple instances simply by double clicking the application.

But I just had a thought like the Mobile Emulators are able to make multiple Instances, if that could be something we could potentially see in future? :D

**Comment by Nightfirecat** [comment]
Jagex offers a Mac client on their download page: https://www.runescape.com/oldschool/download

**Comment by shivharikulhari** [comment]
> Jagex offers a Mac client on their download page: https://www.runescape.com/oldschool/download

Haha thats nice, its not very old because I've been playing Runescape forever, there wasn't one last I checked ( which was very long time ago, I agree )

**Comment by zwilkin** [comment]
`echo
echo "Please enter the number of RuneLite applications that you wish to start."

read NumOfApps

echo "Starting $NumOfApps RuneLite clients..."
echo

for ((i=0; i<NumOfApps; i++)); do
	open -n -a /Applications/RuneLite.app
	echo "Client $((i+1)) is starting up!"
done

exit`

I wrote that for myself. Copy & Paste into a text editor, then save it as a command - like _"MultiRuneLite.command_

**Comment by Hughlav** [comment]
You can make a duplicate (copy + paste) of the runelite applicate and rename it. You can then use that to launch a new instance

**Comment by deathbeam** [comment]
I tried googling, and I have nfc how to force launcher to always open new instance on mac, so I guess we have to stick with just `open -na` unless someone knows (or other solutions people posted here).

**Comment by zwilkin** [comment]
> I tried googling, and I have nfc how to force launcher to always open new instance on mac, so I guess we have to stick with just `open -na` unless someone knows (or other solutions people posted here).

I have development experience specifically in iOS/OSX so before giving up on the feature give me the weekend to do some research into multiple app instances on OSX. 

TBD

**Comment by nlruk** [comment]
> I wrote this for myself when I first encountered this issue. Copy & Paste into a text editor, then save it as a command - like _"MultiRuneLite.**command**"_.
> **MUST BE .command EXTENSION**
> 
> _**EDIT:** Added proper integer check for public-facing use._
> 
> ```
> echo
> echo "Please enter the number of RuneLite applications that you wish to start."
> 
> while true
> do
> 
> read NumOfApps
> 
> if ! [[ "$NumOfApps" =~ ^[0-9]+$ ]]
>     then
> 	echo
>         echo "'$NumOfApps' is invalid."
> 	echo
> 	echo "Please enter a valid number."
>     else
> 	echo "Starting $NumOfApps RuneLite clients..."
> 	echo
> 
> 	for ((i=0; i<NumOfApps; i++)); do
> 		open -n -a /Applications/RuneLite.app
> 		echo "Client $((i+1)) is starting up!"
> 	done
> fi
> 
> done
> exit
> ```
> 
> On startup you will be prompted to enter the number of RuneLite clients you wish to open.

thanks for the code

I've added killall Terminal at the end. do you have a better way to close the terminal after runelite starts up?

**Comment by zwilkin** [comment]
> thanks for the code
> 
> I've added killall Terminal at the end. do you have a better way to close the terminal after runelite starts up?
 
@nlruk 
I wouldn't suggest doing a killall on the Terminal process, but rather changing the _**Ask before closing**_ setting in the Terminal's app preferences...

![image](https://user-images.githubusercontent.com/9438425/79778633-0def0780-8307-11ea-9dcd-67d3c2c071c2.png)
